### PR TITLE
Get block cache usage size

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
1) Get block cache usage size. 
2) `crocksdb_get_options ` not return `rocksdb::Options` for `default` cf but return `rocksdb::DBOptions`, so we implement `get_options` by `get_options_cf`.
@siddontang @BusyJay PTAL

### CODE REVIEW

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -1489,6 +1483,17 @@ void crocksdb_options_set_block_based_table_factory(
   }
 }
 
+size_t crocksdb_options_get_block_cache_usage(crocksdb_options_t *opt) {
+  if (opt) {
+    if (opt->rep.table_factory != nullptr) {
+      void* block_based_table_opt = opt->rep.table_factory->GetOptions();
+      if (block_based_table_opt) {
+        return static_cast<BlockBasedTableOptions*>(block_based_table_opt)->block_cache->GetUsage();
```
Comment (siddontang) [inline librocksdb_sys/crocksdb/c.cc]: do we need to check the table factory is BlockBasedTableFactory first? If not BlockBasedTableFactory, this function may panic.

We can use BlockBasedTableFactory function name() == "BlockBasedTable"  to check it.

**Comment by zhangjinpeng1987** [comment]
@BusyJay PTAL

File: librocksdb_sys/crocksdb/c.cc [hunk 2]
```diff
@@ -1489,6 +1483,17 @@ void crocksdb_options_set_block_based_table_factory(
   }
 }
 
+size_t crocksdb_options_get_block_cache_usage(crocksdb_options_t *opt) {
+  if (opt) {
+    if (opt->rep.table_factory != nullptr) {
+      void* table_opt = opt->rep.table_factory->GetOptions();
+      if (table_opt && strcmp(opt->rep.table_factory->Name(), "BlockBasedTable") == 0) {
```
Comment (siddontang) [inline librocksdb_sys/crocksdb/c.cc]: define a const var here

File: librocksdb_sys/crocksdb/c.cc [hunk 3]
```diff
@@ -1489,6 +1483,17 @@ void crocksdb_options_set_block_based_table_factory(
   }
 }
 
+size_t crocksdb_options_get_block_cache_usage(crocksdb_options_t *opt) {
+  if (opt) {
```
Comment (siddontang) [inline librocksdb_sys/crocksdb/c.cc]: seem we can use `if (opt && opt->rep.table_factory != nullptr)`

File: librocksdb_sys/src/lib.rs [hunk 4]
```diff
@@ -72,8 +72,8 @@ pub enum DBCompactionStyle {
 
 #[repr(C)]
 pub enum DBUniversalCompactionStyle {
-    crocksdb_similar_size_compaction_stop_style = 0,
-    crocksdb_total_size_compaction_stop_style = 1,
+    SimilarSizeCompactionStopStyle = 0,
+    TotalSizeCompactionStopStyle = 1,
```
Comment (BusyJay) [inline librocksdb_sys/src/lib.rs]: Can `CompactionStopStyle` be omitted?

**Comment by zhangjinpeng1987** [comment]
@BusyJay PTAL

File: librocksdb_sys/crocksdb/c.cc [hunk 5]
```diff
@@ -85,6 +85,8 @@ using std::shared_ptr;
 
 extern "C" {
 
+const char* block_base_table_str = "BlockBasedTable";
```
Comment (BusyJay) [inline librocksdb_sys/crocksdb/c.cc]: Is it defined in rocksdb lib already?

Comment (zhangjinpeng1987) [inline librocksdb_sys/crocksdb/c.cc]: Not yet.

**Comment by zhangjinpeng1987** [comment]
@hhkbp2 PTAL

File: tests/test_rocksdb_options.rs [hunk 6]
```diff
@@ -203,3 +203,28 @@ fn test_pending_compaction_bytes_limit() {
...
+        db.put(format!("k_{}", i).as_bytes(), b"v").unwrap();
+    }
+    db.flush(true).unwrap();
+    for i in 0..200 {
+        db.get(format!("k_{}", i).as_bytes()).unwrap();
+    }
+
+    assert!(db.get_options().get_block_cache_usage() > 0);
+
+    let opts = Options::new();
```
Comment (hhkbp2) [inline tests/test_rocksdb_options.rs]: Line 228,229 could be merged to line 211.

**Comment by hhkbp2** [comment]
Rest LGTM

File: tests/test_rocksdb_options.rs (hunk 6 above)
Comment (zhangjinpeng1987) [inline tests/test_rocksdb_options.rs]: This is a special test case that options is empty.

**Comment by siddontang** [comment]
PTAL @hhkbp2

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**